File: src/main/java/com/dennisturco/model/Customer.java
package com.dennisturco.model;

import java.time.LocalDate;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Customer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(optional = false)
    @JoinColumn(name = "customer_type_id", nullable = false)
    private CustomerType customerType;

    @Column(nullable = false, length = 50)
    private String name;

    @Column(nullable = false, length = 50)
    private String surname;

    @Column(nullable = false, unique = true, length = 50)
    private String email;

    @Column(nullable = false, unique = true, length = 20)
    private String phone;

    private String country;

    private String region;

    private String city;

    private String province;

    private String address;

    private String vatNumber;

    private String companyName;

    private String taxCode;

    private String landline;

    private Double lat;

    private Double lon;

    private LocalDate insertDate;

    private LocalDate lastUpdateDate;

    @Column(length = 1000)
    private String notes;
}

File: GestioPro.Common/Models/Customer.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace GestioPro.Common.Models;

[Table("customers")]
public class Customer
{
    [Key]
    public long Id { get; set; }

    [Required]
    public long CustomerTypeId { get; set; }

    [ForeignKey(nameof(CustomerTypeId))]
    public CustomerType CustomerType { get; set; } = null!;

    [Required]
    [MaxLength(50)]
    public string Name { get; set; } = string.Empty;

    [Required]
    [MaxLength(50)]
    public string Surname { get; set; } = string.Empty;

    [Required]
    [MaxLength(50)]
    public string Email { get; set; } = string.Empty;

    [Required]
    [MaxLength(20)]
    public string Phone { get; set; } = string.Empty;

    public string? Country { get; set; }
    public string? Region { get; set; }
    public string? City { get; set; }
    public string? Province { get; set; }
    public string? Address { get; set; }

    public string? VatNumber { get; set; }
    public string? CompanyName { get; set; }
    public string? TaxCode { get; set; }
    public string? Landline { get; set; }

    public double? Lat { get; set; }
    public double? Lon { get; set; }

    public DateOnly InsertDate { get; set; }
    public DateOnly LastUpdateDate { get; set; }

    [MaxLength(1000)]
    public string? Notes { get; set; }
}
